### DISCUSSION TITLE
Video dowload fail everytime with license

### DISCUSSION THREAD

**Discussion by insaf75** [aclap-dev/video-downloadhelper#1150] [Help and Q&A]
### Your question

Hello,

I paid this year a license and I have always been able to download videos from the MyFlixer site but in recent days I must have tried ten times to download a single video but it gives me an error message each time at 8% download.
How to fix the problem ?

### Addon & CoApp version and details

version: 9.1.0.43
target: google
channel: stable
lang: fr
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"cef24b603071428e"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "download_directory": "C:\\Users\\insaf\\dwhelper",
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": false,
    "open_count_store": 10,
    "successfull_dl": 6,
    "dable_attempts": {
        "__serializer_tag": "map",
        "__serializer_value": [
            {
                "__serializer_tag": "array",
                "__serializer_value": [
                    {
                        "__serializer_tag": "primitive",
                        "__serializer_value": "myflixer.pw"
                    },
                    {
                        "__serializer_tag": "array",
                        "__serializer_value": [
                            {
                                "__serializer_tag": "primitive",
                                "__serializer_value": 6
                            },
                            {
                                "__serializer_tag": "primitive",
                                "__serializer_value": 1
                            }
                        ]
                    }
                ]
            }
        ]
    },
    "dable_attempts_last": 1723804312511,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": []
    }
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

Let's have at least one URL of a video that is giving you trouble.  Don't say just that every video is failing.  I'm not going to scrounge around on the site trying to find something that you may or may not have tried to download.  Give me at least one URL of a web page on which I can find a video you have tried unsuccessfully to download.  The purpose of the exercise is to tell me enough that I can try to replicate your results.

**Reply by insaf75** [reply]
I don't understand, it's not the place where we can find help from VDH VAS ?

**Reply by jcfrog** [reply]
This forum is for users and the dev team. 
Everybody can help.

**Reply by GrampaWildWilly** [reply]
Right.  Good help is good help no matter who does the helping.

What is VAS?

**Reply by insaf75** [reply]
https://myflixer.pw/watch-movie/a-hero-76570.8037454

**Reply by GrampaWildWilly** [reply]
Thanks.  I can work with that.

I tried to download this with VDH & I also got only a bit of the movie, 7 minutes of it.  Then VDH failed.  So I went to Plan B.

![#01](https://github.com/user-attachments/assets/e617fa9b-5f84-4254-aad7-6e638b528767)

I undocked the Developer Tools from the video window because it's just more convenient.  What you can see there is an HLS master manifest.  I used that to construct an invocation of ffmpeg.  It started off running fairly fast, about 1 million bytes per second according to the Windows Resource Monitor.  But then it slowed down to about 500 bytes per second.  I don't think I've ever seen a download run that slow, not even on a dial-up modem with a landline back in the bad old days.  But after maybe 5 minutes or so, it did speed back up.  At the moment, it's giving me about 2.5 million bytes per second.  I can't estimate how long this is going to take because I can't predict when it's going to decide to return to glacial speed.  It has swung back & forth wildly while I've been typing this.  It's doing this at the moment:

![#02](https://github.com/user-attachments/assets/279337cc-0b9b-47b4-bd15-5e53502fa5d8)

That's kind of small.  Just click it.  GitHub will open the image in its own browser window, where you can magnify it with the browser's own magnifier (Ctrl-+).

That is simply shocking.  Clearly, the web site is throttling my download.  But I'm not going to closely monitor this.  I'm going to let this run in background while I get on with my life.  When this finishes, and this does give every indication that it will finish within our lifetimes, I will post again with the ffmpeg log.  I'll have more to say then.

**Reply by GrampaWildWilly** [reply]
OK.  It finished.

[Myflixer Movie mp4 Log.txt](https://github.com/user-attachments/files/16640727/Myflixer.Movie.mp4.Log.txt)

The download speed averaged out to be pretty slow but it did download.  A careful scan of the log shows quite a high level of corrupt packets that were dropped.  Nevertheless, I seem to have the complete movie.  I did not sit & watch it, mostly because I'm not interested in it.  Plus, it's in Arabic, a language I don't speak.  I just skimmed it, sampling it at intervals.  There appeared to be both video & audio at every one of my sample points.  I figured out it was Arabic from the credits at the end, which did appear to be almost all there.  You'll note there's about a 3-minute discrepancy between the duration reported in the ffprobe report at the start of that log & the ffprobe report at the end.  All of the Arabic credits were there but after those completed with a copyright notice, it was in the process of displaying some credits in English.  Those looked like they were cut off before they completed.  I have no explanation for that.

I'm a bit surprised there weren't any subtitles offered with this from the manifest.  However, if you look closely at my image of the Developer Tools, you'll see 2 files named eng-2.vtt & eng-3.vtt.  Files of type vtt are usually caption files.  I double clicked the entry for eng-2.vtt & Firefox offered to download the file for me.  I let it, changing the file name to Myflixer Movie.vtt, to match the name of the mp4 I had.  I tested it out & VLC did display English subtitles for this.  Normally, there would be captions mentioned in the master manifest & you could get them with ffmpeg.  But this will do as an alternative method.  It's perhaps a bit non-standard, but it worked.

Commentary . . .

First, I mentioned there were a lot of corrupt packets.  I hope that's not what caused VDH to fail to download this.  There is a parameter for ffmpeg that tells ffmpeg to ignore corrupt packets.  You can see it at the top of the log.  I hope VDH provides this parameter to ffmpeg.

Second, in the ffprobe report near the top of the log you can see there is Timed_ID3 data.  This used to cause VDH problems.  But ffmpeg ignores such data.  I have no idea what such data is used for.  I've never had a tool that would read such data.  I hope that did not cause VDH to fail on this.  VDH is not supposed to have any problems ignoring Timed_ID3 data these days because VDH uses ffmpeg as its download engine.  So this is not supposed to be an issue any longer.

Third, this web site gives quite bad service.  The download speed was intermittently OK, & intermittently shockingly slow.  Plus, there's so many corrupt packets.  Surely you can find a better source for this content.

Bottom line:  I don't see why VDH wouldn't download this.  I used VDH's own ffmpeg to download it & that worked just fine.  What do you say, @paulrouget Paul?
